test/user.test.js

```diff
@@ -1763,6 +1763,54 @@ describe('User', function() {
     });
   });
 
+  describe('password reset with/without email verification', function() {
+    it('allows resetPassword by email if email verification is required and done',
+  function(done) {
+    User.settings.emailVerificationRequired = true;
+    var email = validCredentialsEmailVerified.email;
+    var calledBack = false;
+
+    User.resetPassword({ email: email }, function() {
+      calledBack = true;
+    });
+
+    User.once('resetPasswordRequest', function(info) {
```

Comment:
I am afraid the way how these three new tests are written may be prone to timing-based failures. What if "resetPasswordRequest" gets emitted while `User.resetPassword` is still running?

Since these tests are only verifying whether `User.resetPassword` succeeds or fails, I think there is no need to check "resetPasswordRequest" event.

```js
User.resetPassword({ email: email }, function(err) {
  if (err) return done();
  // success - test has passed
  done();
});
```

The same applies to the third test `it('allows resetPassword by email if email verification is not required'` too.